api/filter_outputs.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+func (api *API) createFilterOutput(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var request createFilterOutputRequest
+	if err := api.ParseRequest(r.Body, &request); err != nil {
+		api.respond.Error(ctx, w, http.StatusBadRequest, errors.Wrap(err, "parse request"))
+		return
+	}
+	if err := api.validate.Struct(&request); err != nil {
+		api.respond.Error(ctx, w, http.StatusBadRequest, errors.Wrap(err, "validate request fields"))
```

Comment:
The request objects are set up to have a valid() function which automatically gets called by parse request, it would be better to include any validation in that function

Reply:
hmmm, let me check

Reply:
done